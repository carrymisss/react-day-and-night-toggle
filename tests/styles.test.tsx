import Adapter from '@cfaester/enzyme-adapter-react-18';
import type { ShallowWrapper } from 'enzyme';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import { describe, expect, test } from 'vitest';

import {
  Circle,
  Cloud,
  Cloudpart,
  Clouds,
  ContentWrapper,
  ShootingStar,
  ShootingStarWrapper,
  Star,
  Stars,
  Switch,
  SwitchContainer,
} from '../src/styles';

Enzyme.configure({ adapter: new Adapter() });

describe("Test of 'styled-components' components:", () => {
  let testSize: number = 32;
  let wrapper: ShallowWrapper<{ className: string; size: number; shadow?: boolean; }>;
  let testShadow: boolean = true;

  test('should <SwitchContainer> render:', () => {
    wrapper = shallow(<SwitchContainer size={testSize} />);
    expect(wrapper).toMatchSnapshot();
    expect((wrapper.props()).size).toEqual(testSize);
  });
  test('should <Switch> render:', () => {
    wrapper = shallow(<Switch size={testSize} shadow={testShadow} />);
    expect(wrapper).toMatchSnapshot();
    const { size, shadow } = wrapper.props();
    expect({ size, shadow }).toEqual({ size: testSize, shadow: testShadow });
  });
  test('should <ContentWrapper> render:', () => {
    wrapper = shallow(<ContentWrapper />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should <Circle> render:', () => {
    wrapper = shallow(<Circle />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should <Star> render:', () => {
    wrapper = shallow(<Star />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should <Stars> render:', () => {
    wrapper = shallow(<Stars />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should <ShootingStar> render:', () => {
    wrapper = shallow(<ShootingStar />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should <ShootingStarWrapper> render:', () => {
    wrapper = shallow(<ShootingStarWrapper />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should <Cloud> render:', () => {
    wrapper = shallow(<Cloud />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should <Clouds> render:', () => {
    wrapper = shallow(<Clouds />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should <Cloudpart> render:', () => {
    wrapper = shallow(<Cloudpart />);
    expect(wrapper).toMatchSnapshot();
  });
});
