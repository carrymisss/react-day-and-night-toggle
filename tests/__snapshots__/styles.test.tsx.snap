// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of 'styled-components' components: should <SwitchContainer> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-aYaIB dZxTuZ",
      "size": 32,
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-aYaIB dZxTuZ",
        "size": 32,
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName size=32 />,
  "length": 1,
}
`;

exports[`Test of 'styled-components' components: should <Switch> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-gEvDqW iTTEjd",
      "shadow": true,
      "size": 32,
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-gEvDqW iTTEjd",
        "shadow": true,
        "size": 32,
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName size=32 shadow=true />,
  "length": 1,
}
`;

exports[`Test of 'styled-components' components: should <ContentWrapper> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-eqUzNf beLJUp",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-eqUzNf beLJUp",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName />,
  "length": 1,
}
`;

exports[`Test of 'styled-components' components: should <Circle> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-fqkwJk ePdnmY",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-fqkwJk ePdnmY",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName />,
  "length": 1,
}
`;

exports[`Test of 'styled-components' components: should <Star> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-iGgVNO jhIIpU",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-iGgVNO jhIIpU",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName />,
  "length": 1,
}
`;

exports[`Test of 'styled-components' components: should <Stars> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-dcJtft",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-dcJtft",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName />,
  "length": 1,
}
`;

exports[`Test of 'styled-components' components: should <ShootingStar> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-kAycRU etyDCI",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-kAycRU etyDCI",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName />,
  "length": 1,
}
`;

exports[`Test of 'styled-components' components: should <ShootingStarWrapper> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-gsFSjX kfhihz",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-gsFSjX kfhihz",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName />,
  "length": 1,
}
`;

exports[`Test of 'styled-components' components: should <Cloud> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-dhKdPU dxxEnp",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-dhKdPU dxxEnp",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName />,
  "length": 1,
}
`;

exports[`Test of 'styled-components' components: should <Clouds> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-jXbVAB",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-jXbVAB",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName />,
  "length": 1,
}
`;

exports[`Test of 'styled-components' components: should <Cloudpart> render: 1`] = `
ShallowWrapper {
  [Symbol(enzyme.__node__)]: {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": {
      "className": "sc-imWZod",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  [Symbol(enzyme.__nodes__)]: [
    {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": {
        "className": "sc-imWZod",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  [Symbol(enzyme.__options__)]: {
    "adapter": ReactEighteenAdapter {
      "options": {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": {
          "componentDidUpdate": {
            "onSetState": true,
          },
          "getChildContext": {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    [Symbol(enzyme.__providerValues__)]: undefined,
  },
  [Symbol(enzyme.__providerValues__)]: Map {},
  [Symbol(enzyme.__renderer__)]: {
    "batchedUpdates": [Function: batchedUpdates],
    "checkPropTypes": [Function: checkPropTypes],
    "getNode": [Function: getNode],
    "render": [Function: render],
    "simulateError": [Function: simulateError],
    "simulateEvent": [Function: simulateEvent],
    "unmount": [Function: unmount],
  },
  [Symbol(enzyme.__root__)]: [Circular],
  [Symbol(enzyme.__unrendered__)]: <NoName />,
  "length": 1,
}
`;
